app/solidus_decorators/spree/variant_decorator.rb

```diff
@@ -3,6 +3,8 @@
 
   private
   def mailchimp_sync
-    SolidusMailchimpSync::VariantSynchronizer.new(self).auto_sync
+    unless product.deleted?
```

Comment:
just curious, is there a way to phrase this with a positive assertion like `if present`